orders/models.py

```diff
@@ -71,3 +73,38 @@ def cancel_order(self):
 
     def get_order_total_price(self):
         return self.price_per_day * (self.end_date - self.start_date).days
+
+    @staticmethod
+    def get_capacity_of_daycare_in_dates_range(daycare_id, start_date, end_date):
+        relevant_orders = Order.objects.filter(daycare_id=daycare_id, status__in=['A', 'O'])
+        start_date = datetime.date(year=start_date.year, month=start_date.month, day=start_date.day)
+        end_date = datetime.date(year=end_date.year, month=end_date.month, day=end_date.day)
+        capacity_per_day_list = [0] * ((end_date - start_date).days + 2)
+
+        for order in relevant_orders:
+            if end_date < order.start_date or order.end_date < start_date:
+                continue
+
+            number_of_days = (order.end_date - order.start_date).days
+
+            for day in range(number_of_days):
+                current_date = order.start_date + datetime.timedelta(days=day)
+                if current_date < start_date:
+                    continue
+                elif current_date > end_date:
+                    break
+                else:
+                    capacity_per_day_list[day] = capacity_per_day_list[day] + 1
+
+        return capacity_per_day_list
+
+    @staticmethod
+    def get_all_day_cares_available_on_dates(start_date: str, end_date: str) -> QuerySet:
+        start_date = datetime.date.fromisoformat(start_date)
```

Comment:
same as I commented on the former function.

Reply:
NP .